### ISSUE TITLE
Problem with big requests (>~100kb)

### ISSUE THREAD

**Issue by bjolletz** [RustedBones/akka-http-metrics#86]
Hi!

After updating to akka-http 10.2.2 and akka-http-metrics 1.4.1, I'm having problems with "big" requests (~> 100kb or so) resulting in 500 internal server error. I have a really hard time tracking down what's really causing this, but my conclusion is that it at least seems to be related to akka-http-metrics. In my experience, the problem arises whenever I add the following import in my Main file where I set up my server.
```
import fr.davit.akka.http.metrics.core.HttpMetrics._
```

I've tried to debug but it seems like the request never even enters the route, so I suspect that something might be crashing pretty early in the request management, resulting in an automatic 500 being returned.

I know this is not much to go on and it's possible this is not a problem with akka-http-metrics at all, but I've been banging my head against the wall with this problem for days now and just curious if there's anyone else seeing the same problem.

I'm using the following versions:
```
akka-http: 10.2.2
akka: 2.5.32
akka-http-metrics: 1.4.1
```

**Comment by RustedBones** [comment]
Hi @bjolletz .
If your server is returning a 500 code, it should produce an error log. Any chance you can access the stacktrace? This will be of great help.

**Comment by bjolletz** [comment]
Hi and thanks for the reply! I apparently didn't have my logging set up correctly, but now when Ive fixed that I can see that I get the following stacktrace:

```
java.lang.IllegalArgumentException: requirement failed: HTTP/1.0 requests must not have a chunked entity
        at scala.Predef$.require(Predef.scala:340)
        at akka.http.scaladsl.model.HttpRequest.<init>(HttpMessage.scala:317)
        at akka.http.scaladsl.model.HttpRequest.copyImpl(HttpMessage.scala:398)
        at akka.http.scaladsl.model.HttpRequest.transformEntityDataBytes(HttpMessage.scala:373)
        at fr.davit.akka.http.metrics.core.HttpMetricsRegistry.onRequest(HttpMetricsRegistry.scala:119)
        at fr.davit.akka.http.metrics.core.HttpMetrics$.$anonfun$meterFunction$2(HttpMetrics.scala:99)
        at fr.davit.akka.http.metrics.core.HttpMetrics$$$Lambda$542/000000007A251850.apply(Unknown Source)
        at scala.Function1.$anonfun$andThen$1(Function1.scala:85)
        at scala.Function1$$Lambda$528/000000007A249540.apply(Unknown Source)
        at scala.Function1.$anonfun$andThen$1(Function1.scala:85)
        at scala.Function1$$Lambda$528/000000007A249540.apply(Unknown Source)
        at scala.Function1.$anonfun$andThen$1(Function1.scala:85)
        at scala.Function1$$Lambda$528/000000007A249540.apply(Unknown Source)
        at akka.stream.impl.fusing.MapAsyncUnordered$$anon$31.onPush(Ops.scala:1376)
        at akka.stream.impl.fusing.GraphInterpreter.processPush(GraphInterpreter.scala:523)
        at akka.stream.impl.fusing.GraphInterpreter.execute(GraphInterpreter.scala:409)
        at akka.stream.impl.fusing.GraphInterpreterShell.runBatch(ActorGraphInterpreter.scala:606)
        at akka.stream.impl.fusing.GraphInterpreterShell$AsyncInput.execute(ActorGraphInterpreter.scala:485)
        at akka.stream.impl.fusing.GraphInterpreterShell.processEvent(ActorGraphInterpreter.scala:581)
        at akka.stream.impl.fusing.ActorGraphInterpreter.akka$stream$impl$fusing$ActorGraphInterpreter$$processEvent(ActorGraphInterpreter.scala:749)
        at akka.stream.impl.fusing.ActorGraphInterpreter$$anonfun$receive$1.applyOrElse(ActorGraphInterpreter.scala:764)
        at akka.actor.Actor.aroundReceive(Actor.scala:539)
        at akka.actor.Actor.aroundReceive$(Actor.scala:537)
        at akka.stream.impl.fusing.ActorGraphInterpreter.aroundReceive(ActorGraphInterpreter.scala:671)
        at akka.actor.ActorCell.receiveMessage(ActorCell.scala:614)
        at akka.actor.ActorCell.invoke(ActorCell.scala:583)
        at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:268)
        at akka.dispatch.Mailbox.run(Mailbox.scala:229)
        at akka.dispatch.Mailbox.exec(Mailbox.scala:241)
        at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
        at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
        at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
        at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
```        
   
From the error message it would sound like I'm sending an invalid (chunked HTTP/1.0) request, but I'm pretty sure I'm sending a HTTP/1.1 request (still need to validate this though). Also, as said above everything seems to work fine if I'm not importing `HttpMetrics._`.

What I could find by googling the error was this:
https://github.com/kamon-io/kamon-akka-http/issues/60

This seems to imply that the problem would have something to do with `transformDataBytes`, which also seems to be called as a result of calling `request.transformEntityDataBytes` in HttpMetricsRegistry:119`.

Thanks for any help looking into this!

**Comment by RustedBones** [comment]
Thanks a lot for your last comment. It has been super helpful!
I could reproduce the bug with a unit test and create a patch.

I've just published it in the `1.4.2-SNAPSHOT` version. Can you give it a try and tell me if this fixes your problem?

**Comment by bjolletz** [comment]
Thanks a lot for the super-quick response. I'd be happy to try it out but not really sure how to get hold of the 1.4.2-SNAPSHOT version. Just changing the version number to `1.4.2-SNAPSHOT` doesn't really seem to work. Also please note that I'm using the Prometheus version.
```
"fr.davit" %% "akka-http-metrics-prometheus" % "1.4.2-SNAPSHOT",
```

Is it published to some specific repository that I need to add in my `build.sbt`?

**Comment by RustedBones** [comment]
Yes, to access snapshot artifacts you must add the following resolver
```sbt
resolvers += Resolver.sonatypeRepo("snapshots")
```
in your `build.sbt`

**Comment by bjolletz** [comment]
I've tried it out and it seems to have solved my problem. Great job! :)

**Comment by bjolletz** [comment]
Will there be a version `1.4.2` released in the near future? I have multiple APIs to update with this fix and I'd rather not depend on a snapshot version if it can be avoided. Thanks again for your great work!

**Comment by RustedBones** [comment]
Just released [`1.5.0`](https://github.com/RustedBones/akka-http-metrics/releases/tag/v1.5.0) (I added a change for `dropwizard` users hence the minor update).

The artifact should be available in a little while.
Thank you for reporting the issue. And sorry for the time lost due to this bug.

**Comment by bjolletz** [comment]
Thanks! The time lost is obviously nothing compared to the effort you're putting into this, so no worries! :) And it's certainly helpful with the very fast response, thanks again for that!
